filters/ema_filter.cpp

```diff
@@ -0,0 +1,105 @@
+#include "ema_filter.hpp"
+
+#include <stout/foreach.hpp>
+
+#include <utility>
+
+namespace mesos {
+namespace serenity {
+
+
+Try<Nothing> EMAFilter::consume(const ResourceUsage& in) {
+  std::unique_ptr<ExecutorSet> newSamples(new ExecutorSet());
+  ResourceUsage product;
+
+  foreach(ResourceUsage_Executor inExec, in.executors()) {
```

Comment:
Personally I would go for 'range-based for loop' from c++11 standard, rather than external libs.